### PR TITLE
Code migration from `AB3` dependencies to `CliMods`

### PR DESCRIPTION
**Pull request by shenyih0ng**
Laying the groundworks for `ModuleList` related commands

**Most of `AB3` related classes are removed so we can declutter the repo and pass compilation!**

## Testing

When the UI loads, it automatically displays the entire list of modules in AY22/23, `UserModuleList` has not been implemented yet so it will be empty.

- Please ignore the failing test cases for now!

## TODOs

- [X] Add global constant for academic year string
- [X] Fix tests

> I am leaning towards not implementing `Storage` for now so we can progress faster for the commands, will do it before `v1.3`

### CODE REVIEW

File: src/main/java/nus/climods/model/module/Module.java [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+        return apiModuleInfo.getSemesterData();
+    }
+
+    public int getModuleCredit() {
+        String moduleCreditStr = apiModuleInfo.getModuleCredit();
+        if (moduleCreditStr.isEmpty()) {
+            return 0;
+        }
+
+        return Integer.parseInt(apiModuleInfo.getModuleCredit());
```
Comment (huzaifa1712) [inline src/main/java/nus/climods/model/module/Module.java:45]: There is a bug here for modules that have a double/float for module credits e.g YSC2251 with 2.5MCs.
Should we make this return a double instead? i.e `Double.parseDouble(apiModuleInfo.getModuleCredit());`

File: src/main/java/nus/climods/ui/module/ModuleCard.java [hunk 2]
```diff
@@ -45,15 +45,22 @@ public class ModuleCard extends UiPart<Region> {
...
-            .forEach(tag -> moduleInfo.getChildren().add(new Pill(tag, SEMESTER_BG_COLOR, SEMESTER_TEXT_COLOR, 13)));
-        moduleInfo.getChildren().add(new Pill(module.getWorkload(), WORKLOAD_BG_COLOR, WORKLOAD_TEXT_COLOR, 13));
+            .forEach(semData -> {
+                assert semData.getSemester() != null;
+                moduleInfo.getChildren()
+                    .add(new Pill(String.format("Semester %s", semData.getSemester().toString()), SEMESTER_BG_COLOR,
+                        SEMESTER_TEXT_COLOR, 13));
+            });
+        moduleInfo.getChildren()
+            .add(new Pill(String.format("%s MCs", module.getModuleCredit()), WORKLOAD_BG_COLOR, WORKLOAD_TEXT_COLOR,
```
Comment (huzaifa1712) [inline src/main/java/nus/climods/ui/module/ModuleCard.java:62]: If we change `getModuleCredit()` to return a double, maybe we could still keep truncation of decimal places for integer MCs like this? Alternatively this code could be moved into the Module class itself to another getter method.
```java
Double moduleCredit = module.getModuleCredit();
String formattedModuleCredit = Math.ceil(moduleCredit) == Math.floor(moduleCredit)
        ? Integer.toString(moduleCredit.intValue()) : Double.toString(moduleCredit);

moduleInfo.getChildren()
    .add(new Pill(String.format("%s MCs", formattedModuleCredit), WORKLOAD_BG_COLOR, WORKLOAD_TEXT_COLOR,
                13));
```

**Review by huzaifa1712 — CHANGES_REQUESTED** [review]
Small bug with getModuleCredit method in Module.java, otherwise LGTM. 
Thanks for cleaning up the repo! 🔥

File: src/main/java/nus/climods/model/module/Module.java (hunk 1 above)
Comment (shenyih0ng) [reply src/main/java/nus/climods/model/module/Module.java:45]: TIL didn't know there's a 2.5 MCs thanks for pointing it out!

**Review by huzaifa1712 — APPROVED** [review]

**Comment by shenyih0ng** [comment]
Academic year can now be set in `preferences.json`!

**Pull request merged**